flink-connector/flink/build.gradle.kts

```diff
@@ -38,19 +38,15 @@ val scalaVersion: String = "2.12"
 val artifactName = "${rootProject.name}-flink-${flinkMajorVersion}_$scalaVersion"
 
 dependencies {
-  implementation(project(":api"))
-  implementation(project(":catalogs:catalog-common"))
-  implementation(project(":common"))
-  implementation(project(":core"))
-
+  // use the client-java instead of client-java-runtime to provide the conflict from
+  // the shaded of com.fasterxml.
+  compileOnly(project(":clients:client-java"))
```

Comment:
Please make those in alphabetical order.

Reply:
Why do we use client-java, not client-java-runtime, the comment doesn't explain clearly.

Reply:
Also, why the scope is "compileOnly"?